plotly/figure_factory/_facet_grid.py

```diff
@@ -331,10 +329,7 @@ def _facet_grid_color_categorical(df, x, y, facet_row, facet_col, color_name,
                                  row_col='row', flipped=flipped_rows)
             )
 
-    # add annotations
-    fig['layout']['annotations'] = annotations
-
-    return fig
+    return fig, annotations
```

Comment:
why was this change made? was it for this? https://github.com/plotly/plotly.py/pull/975/files#diff-83b399f13d8b361d38016264ed1ce589R987 If so, why does `fig['layout']['annotations'].append` not work anymore?

Reply:
Context: Object array properties like (e.g. `layout.annotations`, `layout.images`, etc.) are now stored as tuples instead of lists so that they are immutable.
So `fig['layout']['annotations'].append(annot)` won't work anymore, but `fig['layout']['annotations'] += (annot,)` will work.

Reply:
Thanks Jon for jumping in on Chris' comments. You definitely know the code changes better than I do.

@chriddyp I reworked facet_grid to accommodate annotations being tuples and not lists

Reply:
I see. So if I do 
```
fig = go.Figure(layout=go.Layout(annotations=[go.Annotation(text='...')]))
```
then `annotations` will get cast as a `tuple`.

Marking this one as a 🚨 - Breaking Change